### ISSUE TITLE
Imposters can kill Joker with shield

### ISSUE THREAD

**Issue by gratisCobalt** [NotHunter101/ExtraRolesAmongUs#62]
I don't know if this is really a bug but I didn't found it as a feature so....

If you are Joker you can be killed by an Imposter even if you got a shield.

**Comment by gratisCobalt** [comment]
maybe this is just because of NA servers?

**Comment by nikander100** [comment]
This just happened in our game (will try to get a clip of it happening)

**Comment by rdroog** [comment]
It also happend in our game.

**Comment by nikander100** [comment]
Here is the clip: https://clips.twitch.tv/AlluringFairFlyM4xHeh-7Qy823vOpMbZ-ZUw

**Comment by pobby1092** [comment]
In our round it happened even with a shielded crewmate. My gues is, when 2 imposters are in the area around the shielded person, you kill kill him.

**Comment by benniii1337** [comment]
Yup, same happened in our game today. Shielded as a crewmate, experienced a kill nearby. Was happy to report, but got killed by the 2nd imposter. Nice double for them, but kinda bad for me :cry: 

We guessed some kind of desync behind that.

**Comment by nikander100** [comment]
> In our round it happened even with a shielded crewmate. My gues is, when 2 imposters are in the area around the shielded person, you kill kill him.

Only one impostor pressed kill, i was the other and didnt press kill yet

**Comment by psibean** [comment]
If this really is an issue, the problem is here:
```
ar isPlayerJokerAndCanDieToOfficer =
                        Main.Config.jokerCanDieToOfficer && target.isPlayerRole("Joker");
                    
                    writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
                        (byte) CustomRPC.OfficerKill, Hazel.SendOption.None, -1);
                    writer.Write(PlayerControl.LocalPlayer.PlayerId);
                    writer.Write(!target.isPlayerImmortal() &&
                        (isPlayerJokerAndCanDieToOfficer || target.Data.IsImpostor) ?
                        target.PlayerId : PlayerControl.LocalPlayer.PlayerId); // kill target or suicide
                    AmongUsClient.Instance.FinishRpcImmediately(writer);
                    PlayerControl.LocalPlayer.MurderPlayer(target);
                    PlayerControl.LocalPlayer.getModdedControl().LastAbilityTime = DateTime.UtcNow;

                    // attempted to kill shielded player
                    if (target.isPlayerImmortal())
                    {
                        writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
                            (byte) CustomRPC.AttemptSound, Hazel.SendOption.None, -1);
                        AmongUsClient.Instance.FinishRpcImmediately(writer);
                        BreakShield(false);
                    }
```

There's no `target.isPlayerImmortal` check prior to sending the `OfficerKill`, and the OfficerKill being sent acts out the kill, if this is a bug, then the `breakShield(false)` call in the OfficerKill case should also be removed.

If it is intended that the officer can kill a shielded joker, this should be made clearer in the readme.

**Comment by rdroog** [comment]
This is not related to the officer killing - please read the above comments: an imposter killed a crewmate and joker when that shouldn't be possible. No officer killing other people.

I assume that the target is wrong and that the original target is the closest player around, but that person is another imposter, and then the actual target to be killed isn't checked properly.

**Comment by NotHunter101** [comment]
This is a known bug, and it is not strictly exclusive to the Joker. Refer to pobby1092's comment above for an accurate reason on why it happens. This will be fixed in a future update.
